Show Buildings That Never Submitted

JonHunt1995 opened the pull request:
# Description

Updated data pipeline to no longer filter out buildings that have never reported data to the city. Added `❌` emoji to denote buildings that have never reported data, and added an infographic for building details as well. Finally, added logic to "Out of Date Data" to be mutually exclusive with "Building Never Submitted Data" status.

**Note:** Never submitted buildings show up in search and on the wards pages, which then lets alders see those buildings, and increases the count. E.g. for 40:

| Before | After |
| --- | --- |
| <img width="3016" height="2483" alt="image" src="https://github.com/user-attachments/assets/31f7dd71-4227-4da5-af7f-9e7821f3bf00" /> | <img width="3032" height="2253" alt="image" src="https://github.com/user-attachments/assets/f98dce98-d62e-49c2-a20e-7f87828c46f6" /> |

**Also fixes reporting grade** - some not submitted buildings were getting grades due to being counted as reporting in years with no emissions, so Viktor fixed that as well, closing #288 

| Before | After |
| --- | --- |
| <img width="1383" height="620" alt="image" src="https://github.com/user-attachments/assets/08bf4aba-4c4f-4a30-ba6d-9c28a01fe1df" /> | <img width="1383" height="620" alt="image" src="https://github.com/user-attachments/assets/1dd1d6c8-9af4-4fd0-954c-02379baa6550" /> |


Fixes #170

# Testing Instructions

I looked at a few buildings that had been marked as "❌" to check to see if they don't currently show up with the live website and that the years reported report card matches up with the "Building Never Submitted Data" status.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] If I added a large new feature, I added it to the release notes (ReleaseNotes.vue)

## Data Update (if applicable):

- [ ] I have followed the [Data Update Checklist](DATA_UPDATE_CHECKLIST.md) for updating to new year's data

<!-- PR template modified from: https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ -->

JonHunt1995 committed:
added unsubmitted data

JonHunt1995 committed:
added hasNeverSubmitted getter

JonHunt1995 committed:
Merge branch 'main' into afdsadf
-unsubmitted-data

JonHunt1995 committed:
Added FirstYearReported to page-query for Ward.vue and other pages that use BuildingsTable component

JonHunt1995 committed:
Added Building Never Reported infographic to building details, made out of date data and never reported statuses mutually exclusive

JonHunt1995 commented:
I have just noticed that a lot of the time the "never reported" buildings also commonly flag as "electric only" and I should probably change that so that doesn't happen.

vkoves commented:
@JonHunt1995 - this PR is in draft, but did you mean for me to review it? If so, just hit "Ready for review" at the bottom next time or tag me!

JonHunt1995 committed:
Changed gas free status to filter out nonsubmitting buildings

JonHunt1995 committed:
reverted graphql query

JonHunt1995 committed:
Merge branch 'main' into add-unsubmitted-data

vkoves commented:
Known issues:

- [ ] Search excludes buildings that haven't submitted, but all emissions page shows them
- [ ] Homepage missing buildings that submitted in the past but not recently
- [ ] Need basic UI for never submitted building (e.g. drop all panels and show basic text)

vkoves commented:
@JonHunt1995 - wanted to check in on this, any updates?

vkoves committed:
Merge branch 'main' into add-unsubmitted-data

vkoves committed:
Show Not Submitted Buildings

vkoves committed:
Add never submitted page

vkoves committed:
Improve Not Submitted buildings

vkoves committed:
Don't grade not submitted buildings

vkoves committed:
Fix Pyright error

RamarkoB commented on tests/data/scripts/unit/test_grade_buildings.py:
```diff
@@ -198,6 +222,12 @@ def test_energy_mix_grading():
+    assert pd.isna(id5_row["EnergyMixWeightedPctSum"].iloc[0])
+    assert pd.isna(id5_row["EnergyMixPercentileGrade"].iloc[0])
+    assert pd.isna(id5_row["EnergyMixLetterGrade"].iloc[0])
+
```
Add a one line to check that this find that there is no first year recorded

RamarkoB commented on tests/data/scripts/unit/test_grade_buildings.py:
```diff
@@ -198,6 +222,12 @@ def test_energy_mix_grading():
+    assert pd.isna(id5_row["EnergyMixPercentileGrade"].iloc[0])
+    assert pd.isna(id5_row["EnergyMixLetterGrade"].iloc[0])
+
 
```
Add test to check that `SubmittedRecordsLetterGrade` is `none`, as the corresponds to never submitted

vkoves committed:
Fix map modal with not submitted

vkoves committed:
Hide blank fields from not submitted buildings

vkoves committed:
Add view all not submitted link to building details

vkoves committed:
Add some tests

vkoves committed:
Add Not Submitted to Release Notes

vkoves committed:
Fix Not Submitted buildings on map, show red tooltip

vkoves committed:
Fixed Years Reported Grades
Fixes #288

vkoves committed:
Fix duplicate slugs, showing all buildings
Also made it so a slug error now will break the build, so
we can't just ignore it

vkoves committed:
Fix Reporting Grade Being Artificially High
Have to investigate more since some buildings have no GHG intensity
but have other data

vkoves committed:
Run linting

vkoves committed:
Regenerate Playwright snapshots
